### ISSUE TITLE
Needs to resolve HTML entities

### ISSUE THREAD

**Issue by neil-morgan** [claus/storyblok-rich-text-react-renderer#11]
For example a richtext string like this in storyblok...

<img width="301" alt="Screenshot 2021-11-24 at 21 06 42" src="https://user-images.githubusercontent.com/52212097/143312914-e78e1a62-0fc8-48e5-b72f-9cc1ab227c5e.png">

When resolved like this...
```
[NODE_PARAGRAPH]: children => <Paragraph>{children}</Paragraph>,
```

Will still render the html entity...

<img width="894" alt="Screenshot 2021-11-24 at 21 07 42" src="https://user-images.githubusercontent.com/52212097/143312985-ce32c134-3ea7-46cc-a79e-232acf36e614.png">

**Comment by claus** [comment]
Hey @neil-morgan, i think it would be a bit overkill to add a full HTML entity resolver to this lib, but what i could do is expose a `textResolver` via options, that gets called on all text nodes and that by default does nothing. You could then hook into that and run e.g. [html-entities](https://www.npmjs.com/package/html-entities)' `decode()` function on. Would that help?

**Comment by neil-morgan** [comment]
@claus if I am understanding you correctly I think that would be a massive help.

Essentially in an example like this:

```
        render(content.body, {
          nodeResolvers: {
            [NODE_PARAGRAPH]: children => <Paragraph>{children}</Paragraph>,
            [NODE_UL]: children => <List>{children}</List>,
            [NODE_OL]: children => (
              <OrderedList component="ol">{children}</OrderedList>
            ),
            [NODE_LI]: children => <ListItem>{children}</ListItem>
          },
          markResolvers: {
            [MARK_LINK]: (children, props) => (
              <Link href={props.href}>{children}</Link>
            )
          }
        })
```

It would be great to be able to resolve HTML entities wherever {children} is rendered.

**Comment by claus** [comment]
Yeah there would be a new `textResolver`, and all you'd have to do is:
```js
render(doc, {
    textResolver: (string) => decode(string)
})
```
I haven't tested this yet, will do over the long weekend.

**Comment by neil-morgan** [comment]
If I am getting this right would you then have something like the below?

```
  render(doc, {
    textResolver: string => decode(string),
    nodeResolvers: {
      [NODE_PARAGRAPH]: children => <Paragraph>{children}</Paragraph>
    }
  });
```

**Comment by claus** [comment]
Yes, exactly

**Comment by claus** [comment]
@neil-morgan I published 2.5.0 with the new `textResolver` option, let me know if that works for you.

**Comment by neil-morgan** [comment]
@claus works perfectly. Thank you very much for your effort on this and the project as a whole.
